src/network/protocol/redis/command/network_protocol_redis_command_mget.c

```diff
@@ -0,0 +1,316 @@
+/**
+ * Copyright (C) 2018-2022 Vito Castellano
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the BSD license.  See the LICENSE file for details.
+ **/
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include <strings.h>
+#include <arpa/inet.h>
+#include <assert.h>
+
+#include "misc.h"
+#include "exttypes.h"
+#include "log/log.h"
+#include "clock.h"
+#include "spinlock.h"
+#include "data_structures/small_circular_queue/small_circular_queue.h"
+#include "data_structures/double_linked_list/double_linked_list.h"
+#include "data_structures/queue_mpmc/queue_mpmc.h"
+#include "slab_allocator.h"
+#include "data_structures/hashtable/mcmp/hashtable.h"
+#include "data_structures/hashtable/mcmp/hashtable_op_get.h"
+#include "protocol/redis/protocol_redis.h"
+#include "protocol/redis/protocol_redis_reader.h"
+#include "protocol/redis/protocol_redis_writer.h"
+#include "network/protocol/network_protocol.h"
+#include "network/io/network_io_common.h"
+#include "config.h"
+#include "fiber.h"
+#include "network/channel/network_channel.h"
+#include "storage/io/storage_io_common.h"
+#include "storage/channel/storage_channel.h"
+#include "storage/db/storage_db.h"
+#include "network/protocol/redis/network_protocol_redis.h"
+#include "network/network.h"
+#include "worker/worker_stats.h"
+#include "worker/worker_context.h"
+
+#define TAG "network_protocol_redis_command_mget"
+
+struct mget_command_key_data {
+    char *key;
+    size_t key_length;
+    size_t key_offset;
+};
+typedef struct mget_command_key_data mget_command_key_data_t;
+
+struct mget_command_context {
+    char error_message[200];
+    bool has_error;
+    mget_command_key_data_t *key_data;
+    mget_command_key_data_t **keys;
+    uint keys_count;
+};
+typedef struct mget_command_context mget_command_context_t;
+
+NETWORK_PROTOCOL_REDIS_COMMAND_FUNCPTR_COMMAND_BEGIN(mget) {
+    protocol_context->command_context = slab_allocator_mem_alloc(sizeof(mget_command_context_t));
+
+    mget_command_context_t *mget_command_context = (mget_command_context_t*)protocol_context->command_context;
+    mget_command_context->has_error = false;
+    mget_command_context->keys_count = reader_context->arguments.count - 1;
+    mget_command_context->keys = slab_allocator_mem_alloc(mget_command_context->keys_count * sizeof(mget_command_key_data_t*));
```

Comment:
Better to zero this memory upon allocation, it makes easy to identify that there is memory to be freed in NETWORK_PROTOCOL_REDIS_COMMAND_FUNCPTR_COMMAND_FREE (there is a related comment there).